Jellyfin.Api/Controllers/UserLibraryController.cs

```diff
@@ -381,5 +385,42 @@ private UserItemDataDto UpdateUserItemRatingInternal(Guid userId, Guid itemId, b
 
             return _userDataRepository.GetUserDataDto(item, user);
         }
+
+        /// <summary>
+        /// Gets an item's lyrics.
+        /// </summary>
+        /// <param name="userId">User id.</param>
+        /// <param name="itemId">Item id.</param>
+        /// <response code="200">Lyrics returned.</response>
+        /// <response code="404">Something went wrong. No Lyrics will be returned.</response>
+        /// <returns>An <see cref="OkResult"/> containing the intros to play.</returns>
+        [HttpGet("Users/{userId}/Items/{itemId}/Lyrics")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<QueryResult<BaseItemDto>> GetLyrics([FromRoute, Required] Guid userId, [FromRoute, Required] Guid itemId)
+        {
+            var user = _userManager.GetUserById(userId);
+
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            var item = itemId.Equals(default)
+                ? _libraryManager.GetUserRootFolder()
+                : _libraryManager.GetItemById(itemId);
+
+            if (item == null)
+            {
+                return NotFound();
+            }
+
+            List<Lyrics> result = ItemHelper.GetLyricData(item);
+            if (string.IsNullOrEmpty(result.ElementAt(0).Error))
+            {
+                return Ok(new { Results = result });
+            }
```

Comment:
At minimum I would suggest an ILyricProvider and a lambda here that searches through existing implementations to find the first result. Bonus points if there's a setting somewhere to select a particular lyric source, but as long as the provider exists that can always be added later.

Reply:
You'll have to forgive me, C# isn't my primary language. I think I'm getting what you're saying here, but if I have any followup questions, do you mind if I reach out for clarification?

Reply:
No problem, you can find me on Matrix. For the record, the IDv3 lyric support mentioned by @ferferga is a great example of code that would go in a separate LyricProvider down the road. People could even implement new providers through plugins if they want support for a paid service or something.